### PR TITLE
erc20 xtoken transfer

### PR DESCRIPTION
**Pull request by zqhxuyuan**
orml companion: https://github.com/open-web3-stack/open-runtime-module-library/pull/752
closes: https://github.com/AcalaNetwork/Acala/issues/2167

- [x] support ERC20 withdraw and deposit, as evm not support it, we're using `Erc20HoldingAccount` to transfer.
- [x] rename `FixedRateOfForeignAsset` to `FixedRateOfAssetRegistry`, and support both `ForeignAsset` and `Erc20` 
    - introduce `BuyWeightRate` to calculate rate by location.
- [x] testcase for Erc20 xcm transfer
    - Karura send USDC to Sibling with location: (1, 2000, GeneralKey(Ecr20(evm_address))). Sibling will take it as ForeignAsset.
    - Sibling send back USDC to Karura with location (1, 2000, GeneralKey(Ecr20(evm_address))), then in Karura, it'll be (0, GeneralKey(Ecr20(evm_address))), and it'll be consider as Erc20 token
    - Sibling send USDC to Karura with location: (1, 2001, GeneralKey(Ecr20(evm_address))). Karura will take it as ForeignAsset.

### CODE REVIEW

**Review by xlc — DISMISSED** [review]

File: runtime/acala/src/lib.rs [hunk 1]
```diff
@@ -839,13 +839,15 @@ parameter_types! {
 	pub const GetStableCurrencyId: CurrencyId = AUSD;
 	pub const GetLiquidCurrencyId: CurrencyId = LDOT;
 	pub const GetStakingCurrencyId: CurrencyId = DOT;
+	pub AccountForTransfer: AccountId = AccountId::new([0u8; 32]);
```
Comment (zjb0807) [inline runtime/acala/src/lib.rs:842]: We should avoid using the 0 account id, it is not secure.
We can use `NetworkContractSource`

File: modules/currencies/src/lib.rs [hunk 2]
```diff
@@ -90,6 +90,9 @@ pub mod module {
 		#[pallet::constant]
 		type GetNativeCurrencyId: Get<CurrencyId>;
 
+		/// Used as temporary account for ERC20 token `withdraw` and `deposit`.
+		type AccountForTransfer: Get<Self::AccountId>;
```
Comment (xlc) [inline modules/currencies/src/lib.rs:94]: I don't like this name. Maybe Erc20HoldingAccount?

File: runtime/acala/src/lib.rs (hunk 1 above)
Comment (zqhxuyuan) [reply runtime/acala/src/lib.rs]: `NetworkContractSource` in the runtime is `H160::from_low_u64_be(0)`, it'll be error when trying send to it. can we use `H160::from_low_u64_be(1)`?

File: runtime/karura/src/lib.rs [hunk 3]
```diff
@@ -854,13 +854,15 @@ parameter_types! {
 	pub const GetStableCurrencyId: CurrencyId = KUSD;
 	pub const GetLiquidCurrencyId: CurrencyId = LKSM;
 	pub const GetStakingCurrencyId: CurrencyId = KSM;
+	pub Erc20HoldingAccount: H160 = H160(hex_literal::hex!("000000000000000000ff00000000000000000000"));
```
Comment (shaunxw) [inline runtime/karura/src/lib.rs:857]: This address looks pretty arbitrary to me. Can we use an account derived from currencies module pallet id?

Comment (xlc) [reply runtime/karura/src/lib.rs:857]: we want to use a system account, that's the one starts `000000000000000000`

I think we want to have this defined around here https://github.com/AcalaNetwork/Acala/blob/ded6de57234c4367401dbd758db609254c2e00e0/primitives/src/evm.rs#L129-L153

**Review by xlc — DISMISSED** [review]

**Review by shaunxw — DISMISSED** [review]

**Review by zjb0807 — APPROVED** [review]

**Pull request merged**